src/shogun/lib/observers/ObservedValueTemplated.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Giovanni De Toni
+ */
+
+#ifndef SHOGUN_OBSERVEDVALUETEMPLATED_H
+#define SHOGUN_OBSERVEDVALUETEMPLATED_H
+
+#include <shogun/lib/observers/ObservedValue.h>
+
+namespace shogun {
+
+	/**
+ 	* Templated specialisation of ObservedValue that stores the actual data.
+ 	* @tparam T the type of the observed value
+ 	*/
+	template<class T>
+	class ObservedValueTemplated : public ObservedValue {
```

Comment:
I am not sure I understand why you have a templated class though? Why do you need to have a template and the Any value? I think I am missing something :)

Reply:
The templated version is needed to be able to register the observed value within the parameter framework. The `Any` value is needed because there are situations in which we need to use the `sg_any_dispatch` method to retrieve the observed value. Therefore I had to add both functionalities to the classes :)

Reply:
ah ok yes, I see!